qstack/spahm/rho/bond.py

```diff
@@ -10,53 +10,47 @@
 def bond(mols, dms,
          bpath=defaults.bpath, cutoff=defaults.cutoff, omods=defaults.omod,
          spin=None, elements=None, only_m0=False, zeros=False, split=False, printlevel=0,
-         pairfile=None, dump_and_exit=False):
+         pairfile=None, dump_and_exit=False, same_basis=False, only_z=[]):
 
     elements, mybasis, qqs0, qqs4q, idx, M = dmbb.read_basis_wrapper(mols, bpath, only_m0, printlevel,
                                                                      elements=elements, cutoff=cutoff,
-                                                                     pairfile=pairfile, dump_and_exit=dump_and_exit)
+                                                                     pairfile=pairfile, dump_and_exit=dump_and_exit, same_basis=same_basis)
     if spin is None:
         omods = [None]
     qqs = qqs0 if zeros else qqs4q
     maxlen = max([dmbb.bonds_dict_init(qqs[q0], M)[1] for q0 in elements])
-    natm   = max([mol.natm for mol in mols])
+    if len(only_z) > 0:
+        print(f"Selecting atom-types in {only_z}")
+        zinmols = []
+        for mol in mols:
+            zinmol = [sum(z == np.array(mol.elements)) for z in only_z]
+            zinmols.append(sum(zinmol))
+        natm  = max(zinmols)
+    else:
+        natm   = max([mol.natm for mol in mols])
+        zinmols = [mol.natm for mol in mols]
     allvec = np.zeros((len(omods), len(mols), natm, maxlen))
 
     for imol, (mol, dm) in enumerate(zip(mols,dms)):
         if printlevel>0: print('mol', imol, flush=True)
         for iomod, omod in enumerate(omods):
             DM  = utils.dm_open_mod(dm, omod) if spin else dm
-            vec = dmbb.repr_for_mol(mol, DM, qqs, M, mybasis, idx, maxlen, cutoff)
+            vec = dmbb.repr_for_mol(mol, DM, qqs, M, mybasis, idx, maxlen, cutoff, only_z=only_z)
             allvec[iomod,imol,:len(vec)] = vec
 
     if split is False:
         shape  = (len(omods), -1, maxlen)
-        atidx  = np.where(np.array([[1]*mol.natm + [0]*(natm-mol.natm) for mol in mols]).flatten())
+        atidx  = np.where(np.array([[1]*zin + [0]*(natm-zin) for zin in zinmols]).flatten())
         allvec = allvec.reshape(shape)[:,atidx,:].reshape(shape)
     return allvec
-
-
-def get_repr(mols, xyzlist, guess,  xc=defaults.xc, spin=None, readdm=None,
-             pairfile=None, dump_and_exit=False,
-             bpath=defaults.bpath, cutoff=defaults.cutoff, omods=defaults.omod,
-             elements=None, only_m0=False, zeros=False, split=False, printlevel=0):
-
-    dms     = utils.mols_guess(mols, xyzlist, guess,
-                               xc=xc, spin=spin, readdm=readdm, printlevel=printlevel)
-    allvec  = bond(mols, dms, bpath, cutoff, omods,
-                   spin=spin, elements=elements,
-                   only_m0=only_m0, zeros=zeros, split=split, printlevel=printlevel,
-                   pairfile=pairfile, dump_and_exit=dump_and_exit)
-    return allvec
```

Comment:
I'm not sure what this function is used for ?

_maybe a left-over from before `bond()`??_

Reply:
Look in the master branch `main()` calls `get_repr()` and `get_repr()` calls `bond()`.
I think the idea was to have an inner function that doesn't compute the guess, just the representation.
I would rather split spahm(a) in the same manner